app/models/report.rb

```diff
@@ -416,10 +421,12 @@ def pivots
   end
 
   def pivot_fields
-    @pivot_fields ||= Field.find_by_name(pivots)
-      .group_by{|f| f.name}
-      .map{|k,v| [k, v.first]}
-      .to_h
+    @pivot_fields ||= pivots.present? ?
```

Comment:
Is adding the pivots.present causing the pivots calculation to be executed twice here?   If so, I'd recommend improving that method to only execute once.  
 Example:
        @pivots ||= (self.aggregate_by || []) + (self.disaggregate_by || [])